pyccel/ast/cwrapper.py

```diff
@@ -134,9 +134,18 @@ class PyArg_ParseTupleNode(Basic):
         List of arguments into which the result will be collected
     arg_names : list of str
         A list of the names of the function arguments
+    is_interface : boolean
+        Default value False and True when working with interface functions
     """
 
-    def __init__(self, python_func_args, python_func_kwargs, c_func_args, parse_args, arg_names):
+    def __new__(cls, *args, **kwargs):
+        return Basic.__new__(cls, *args, **kwargs)
```

Comment:
can you explain

Reply:
When a class inherits from another class (here `PyArg_ParseTupleNode` inherits from `Basic`) it inherits all of its methods. So when we call `__new__` if we haven't redefined `__new__` then we are calling `Basic.__new__`.

Sometimes we need to implement `__new__` ourselves. For example if the arguments to `__new__` for the base class don't match the arguments for the sub-class or if we want to leave ourselves the possibility to return a different type.

This is not the case here as you are calling `Basic.__new__` with the same arguments that were passed to `PyArg_ParseTupleNode.__new__`. This means that the same set of steps would be carried out whether this function is here or not.

Reply:
If you want a refresher on classes and inheritance it might be useful to look at `ast.operators` and maybe pick issue #564 when it appears in a sprint as there is lots of inheritance there. The inheritance for the `__new__` and `__init__` methods work in the same way as the inheritance for all the other methods with the exception that the first argument of `__new__` is a type instead of an instance of the class

Reply:
This comment seems to have been lost, but it is still valid